### PR TITLE
Added bindings to ESP-IDF native filesystem library for ugfx via USERFS

### PR DESCRIPTION
**Pull request by aczid**
With this patch UGFX can finally see the regular filesystem.
Also requires https://github.com/SHA2017-badge/micropython-esp32/pull/130

### CODE REVIEW

File: components/ugfx-glue/gfx_userfs.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    f->flags |= GFILEFLG_OPEN;
+    return f->obj != NULL;
+}
+
+void userfs_close(GFILE *f){
+    fclose(f->obj);
+}
+
+int userfs_read(GFILE *f, void * buf, int size){
+    return fread(buf, 1, size, f->obj);
```
Comment (basvs) [inline components/ugfx-glue/gfx_userfs.c]: Add check if f->obj is set?

File: components/ugfx-glue/gfx_userfs.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+int userfs_write(GFILE *f, const void *buf, int size){
+    return fwrite(buf, 1, size, f->obj);
```
Comment (basvs) [inline components/ugfx-glue/gfx_userfs.c]: Add check if f->obj is set?

File: components/ugfx-glue/gfx_userfs.c [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+bool_t userfs_setpos(GFILE *f, long int pos){
+    return fseek(f->obj, pos, SEEK_SET) != -1;
```
Comment (basvs) [inline components/ugfx-glue/gfx_userfs.c]: Add check if f->obj is set?

File: components/ugfx-glue/gfx_userfs.c [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+long int userfs_getsize(GFILE * f){
+    long int oldpos = ftell(f->obj);
```
Comment (basvs) [inline components/ugfx-glue/gfx_userfs.c]: Add check if f->obj is set?

File: components/ugfx-glue/gfx_userfs.c [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    fseek(f->obj, 0, SEEK_END);
+    long int size = ftell(f->obj);
+    fseek(f->obj, oldpos, SEEK_SET);
+    return size;
+}
+
+bool_t userfs_eof(GFILE *f){
+    return feof(f->obj);
```
Comment (basvs) [inline components/ugfx-glue/gfx_userfs.c]: Add check if f->obj is set?

File: components/ugfx-glue/gfx_userfs.c [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+            f->obj = fopen(fname, "rw");
+            break;
+            /*return 0;*/
+    }
+    f->flags |= GFILEFLG_OPEN;
+    return f->obj != NULL;
+}
+
+void userfs_close(GFILE *f){
+    fclose(f->obj);
```
Comment (basvs) [inline components/ugfx-glue/gfx_userfs.c]: Maybe remove the GFILEFLG_OPEN from f->flags

**Comment by aczid** [comment]
Thank you for keeping the quality up. :)

**Comment by basvs** [comment]
Thank you! :)

**Comment by basvs** [comment]
Argh, missed that one. Now wait for Travis...

**Pull request merged**